runtime/parachains/src/paras_inherent.rs

```diff
@@ -252,25 +256,24 @@ pub mod pallet {
 			// (`enter`) and the off-chain checks by the block author (this function). Once we are confident
 			// in all the logic in this module this check should be removed to optimize performance.
 
-			let inherent_data =
-				match Self::enter(frame_system::RawOrigin::None.into(), inherent_data.clone()) {
-					Ok(_) => inherent_data,
-					Err(err) => {
-						log::error!(
-							target: LOG_TARGET,
-							"dropping paras inherent data because they produced \
+			let inherent_data = match Self::enter_inner::<true>(inherent_data.clone()) {
```

Comment:
A comment on the fact that we skip inherent checks here would be good - we are essentially assuming that the node-side code hasn't made an error.

Reply:
(actually, I now see that `create_inherent_inner` is doing those checks. but yeah, still worth a comment on why we skip that.)

Reply:
Not exactly. We're assuming that the sanitization worked, which for the `inherent_filter` already does these checks - we just don't need them to be done twice and risk missing our slot.